src/project/state.js

```diff
@@ -0,0 +1,28 @@
+export const state = {
+  projects: [],
+  selectedProjectId: undefined,
+};
+
+export const addProject = (project) => {
+  state.projects.push(project);
+};
+
+export const selectProject = (project) => {
+  const index = state.projects.findIndex((p) => p === project);
+
+  state.selectedProjectId = index;
+};
+
+export const getSelectedProject = () => {
+  if (state.selectedProjectId === undefined) {
+    return;
+  }
+
+  return state.projects[state.selectedProjectId];
+};
+
+export const deleteProject = (project) => {
+  const index = state.projects.find((p) => p === project);
+
+  state.projects.splice(index, 1);
+};
```

Comment:
Having a bunch of method to mutate the state is a good idea. It regroups the different actions that can happen in your application.

Here we can see that you can add a project, select a project, delete a project, and retrieve the selected project.

These are helpers to use elsewhere.